src/site/content/en/reliable/two-way-communication-guide/index.md

```diff
@@ -0,0 +1,400 @@
+---
+layout: post 
+title: "Two way communication" 
+authors:
+  - demianrenzulli
+  - andrewguan 
+date: 2020-12-04 
+description: | 
+    How establish a two-way communication channel between the page and the service worker. 
+tags:
+  - blog
+  - service-worker
+  - web-worker
+---
+
+In some cases, a web app might need to establish a **two-way** communication channel between the
+page and the service worker.
+
+For example: in a podcast PWA one could build a feature to let the user [download episodes for
+offline consumption](https://web.dev/app-like-pwas/#proactive-background-downloading) and allow the
+service worker to keep the page regularly informed about the progress, so the [main
+thread](https://developer.mozilla.org/en-US/docs/Glossary/Main_thread) can update the UI.
+
+In this guide we'll explore the different ways of implementing a **two-way** communication between
+the [Window](https://developer.mozilla.org/en-US/docs/Web/API/Window) and [service
+worker](https://developer.mozilla.org/en-US/docs/Web/API/Service_Worker_API) context, by exploring
+different APIs, the [Workbox library](https://developers.google.com/web/tools/workbox), as well as
+some advanced cases.
+
+<figure class="w-figure">
+  <img src="two-way-communication-diagram.png"
+       width="800"
+       alt="Diagram showing a service worker and the page exchanging messages.">
+</figure>
```

Comment:
Is this diagram complete? 

![Uploading image.png…]()

Reply:
It's the diagram that Andrew used in this part of the the talk, so, yes. Maybe it doesn't have text as in others. Do you think it's necessary?